lib/matplotlib/axes/_base.py

```diff
@@ -3073,15 +3074,18 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
         >>> set_xlim(5000, 0)
 
         """
-        if 'xmin' in kw:
-            left = kw.pop('xmin')
-        if 'xmax' in kw:
-            right = kw.pop('xmax')
-        if kw:
-            raise ValueError("unrecognized kwargs: %s" % list(kw))
-
         if right is None and iterable(left):
             left, right = left
+        if xmin is not None:
+            if left is not None:
+                raise ValueError('Cannot use argument xmin=%r; would '
```

Comment:
This technically breaks backwards compatibility I think - previously passing `xmin` and `left` would silently use the value in `xmin`, whereas now it raises an error.

Reply:
Yes. Should just issue a warning. But what to be done in this context depends on the decision on deprecation.